cmake/git.cmake

```diff
@@ -1,6 +1,20 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# Use git if it is installed, to set BUILD_VERSION
+#.rst:
+# git.cmake
+# ---------
+# If the user didn't already define BUILD_VERSION then try to initialize
+# it with the output of "git describe". Warn but don't error if
+# everything fails and leave BUILD_VERSION undefined.
+#
+# Inputs:
+#   ``BUILD_VERSION`` or the output of ``git describe``
```

Comment:
How is

the output of ``git describe``

an input to git.cmake ?

Reply:
The output of git describe is not an input to git.cmake, it is an internal process of git.cmake.

If we had run git describe before git.cmake, then you could say it was an input.

Reply:
I don't see why who runs git describe and when matters. The terms "input" and "output" are standard for  internal calls too:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out-parameter-modifier
https://docs.swift.org/swift-book/LanguageGuide/Functions.html
https://www.learncpp.com/cpp-tutorial/73-passing-arguments-by-reference/
etc.

In fact CMake's documentation for execute_process() uses the terms "input" and "output" too:
https://cmake.org/cmake/help/v3.14/command/execute_process.html?highlight=output

Reply:
I find the documentation causes more confusion than than it clarifies, but perhaps this is subjective.

Perhaps "the output of 'git describe'" will clarify things for some users.

This is OK.

Reply:
I wrote the "inputs" and "outputs" section in git.cmake for consistency with version.cmake but now I see they've become mostly redundant with the plain English sentence just before. I'll drop them; less is more. I assume the first sentence is OK.
The last sentence with the reference to the (independent!) KERNEL_VERSION_* is the most important IMHO